src/hotspot/share/opto/compile.cpp

```diff
@@ -3657,7 +3660,10 @@ void Compile::final_graph_reshaping_main_switch(Node* n, Final_Reshape_Counts& f
       } else if (t->isa_oopptr()) {
         n->subsume_by(ConNode::make(t->make_narrowoop()), this);
       } else if (t->isa_klassptr()) {
-        n->subsume_by(ConNode::make(t->make_narrowklass()), this);
+        ciKlass* klass = t->is_klassptr()->exact_klass();
```

Comment:
This branch means that we are trying to compress a pointer that cannot be compressed. This seems wrong either way.

Reply:
I think it's always bad if we reach here with an abstract or interface class. `ConP` + `EncodeP` will still be wrong. Usage of the result can cause strange issues.
Can we assert that `klass->is_in_encoding_range()`, here?

Reply:
All right, I agree with all three of you. New commit adds the assert and bailout when we detect that we are encounter the unencodable class in this `ConP` -> `EncodeP` path. Does that make more sense to you?